cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -527,7 +561,7 @@ struct MatrixLinewiseOp {
                   const bool alongLines,
                   Lambda op,
                   cudaStream_t stream,
-                  Vecs... vecs)
+                  const Vecs*... vecs)
```

Comment:
Would you consider splitting the matrix element `Type` into `InT`/`OutT` types for bonus points? I guess, the only thing we'd need to change in the logic then is how we check the alignment requirements for input/output.

Reply:
As it complicates the alignment logic a bit and we haven't seen a case yet where this is needed, I think it's better to keep that in the backlog.